### PR TITLE
#14628: Remove ARCH_NAME specific parameter from kernel.cpp

### LINKED ISSUE #14628: kernel.hpp uses ARCH_NAME specific include path to provide ERISC_L1_KERNEL_CONFIG_SIZE at compile time
https://github.com/tenstorrent/tt-metal/blob/97731c4e64db1306a751ddc467e6e1e65f1009f7/tt_metal/impl/kernels/kernel.hpp#L30

This parameter is already available behind the hal, but the hal cannot be queried at compile time.

This code needs restructuring to check this parameter at runtime.

### PR DESCRIPTION
**Pull request by blozano-tt**
### Ticket
Closes #14628
#596 

### Problem description
Existing code had compile time dependence on `ARCH_NAME`.

### What's changed
Get the parameter from hal instead of at compile time.

### Checklist
https://github.com/tenstorrent/tt-metal/actions/runs/11700899822
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/impl/kernels/kernel.cpp [hunk 1]
```diff
@@ -213,6 +213,7 @@ RuntimeArgsData &Kernel::common_runtime_args_data() { return this->common_runtim
 void Kernel::validate_runtime_args_size(
     size_t num_unique_rt_args, size_t num_common_rt_args, const CoreCoord &logical_core) {
     uint32_t total_rt_args = (num_unique_rt_args + num_common_rt_args);
+    uint32_t idle_eth_max_runtime_args = hal.get_dev_size(HalProgrammableCoreType::ACTIVE_ETH, HalL1MemAddrType::KERNEL_CONFIG) / sizeof(uint32_t);
```
Comment (blozano-tt) [inline tt_metal/impl/kernels/kernel.cpp:216]: @abhullar-tt could you please double check this for me?

My change here is a brain dead substitution, however its bizarre to me that the variable was named "idle_eth_*" and I have to replace with a parameter defined as part of "ACTIVE_ETH" core.

Comment (blozano-tt) [reply tt_metal/impl/kernels/kernel.cpp:216]: Hmmm... current change causes segmentation fault in all grayskull test cases.

Comment (blozano-tt) [reply tt_metal/impl/kernels/kernel.cpp]: I think I fixed ... grayskull doesn't have a hal entry for active ethernet core.

File: tt_metal/impl/kernels/kernel.cpp [hunk 2]
```diff
@@ -213,6 +213,8 @@ RuntimeArgsData &Kernel::common_runtime_args_data() { return this->common_runtim
 void Kernel::validate_runtime_args_size(
     size_t num_unique_rt_args, size_t num_common_rt_args, const CoreCoord &logical_core) {
     uint32_t total_rt_args = (num_unique_rt_args + num_common_rt_args);
+    auto arch = hal.get_arch();
```
Comment (abhullar-tt) [inline tt_metal/impl/kernels/kernel.cpp:216]: so we have a way to pull the arch out of tt::Cluster, don't think we need to duplicate that

need a separate discussion of what lives in Cluster and what gets pulled out of the HAL. This can be finetuned in separate PR

Comment (blozano-tt) [reply tt_metal/impl/kernels/kernel.cpp:216]: Last night it made sense to me that we should be able to ask the Hal what arch it is abstracting.

Comment (abhullar-tt) [reply tt_metal/impl/kernels/kernel.cpp:216]: yeah I do agree. I guess its okay there are multiple ways to access arch

**Review by abhullar-tt — APPROVED** [review]
